### PR TITLE
Caused by android.database.sqlite.SQLiteException: Expression tree is too large (maximum depth 1000) (code 1)

### LINKED ISSUE #1231: Caused by android.database.sqlite.SQLiteException: Expression tree is too large (maximum depth 1000) (code 1)
Logs from firebase:

> Caused by android.database.sqlite.SQLiteException: Expression tree is too large (maximum depth 1000) (code 1): , while compiling: SELECT canEditWhenComplete, deletedDate, status, jrVersion, jrFormId, instanceFilePath, date, submissionUri, displayName, _id, displaySubtext FROM instances WHERE (_id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or _id=? or 

It occurs when you try to send lots of finalized forms to google sheet.

### PR DESCRIPTION
**Pull request by grzesiek2010**
This PR is in reference to resolve #1231

### CODE REVIEW

**Comment by grzesiek2010** [comment]
@mmarciniak Please test this pr really carefully:
-sending forms to Aggregate
-sending forms to Google Sheet (using new Google account or one with removed creditionals as well)
-autosend option to confirm that regression doesn't occur

**Comment by mmarciniak90** [comment]
Tested with success. 
Tested on Android 4.1, 5.1, 6.0
* Aggregate - I did not notice regression. User is able to send 1000 forms during one upload
* Google Sheet - Crash does not occur. User is able to send 1000 forms during one upload
* ODK Collect is not connected to Google account - User sees request to allow to connect app
screenshot_20170718-140551
* Autosend option - I did not notice regression
Data needed to test issue: [backup1249.zip](https://github.com/opendatakit/collect/files/1155963/backup1249.zip)

File: collect_app/src/main/java/org/odk/collect/android/tasks/InstanceGoogleSheetsUploader.java [hunk 1]
```diff
@@ -107,7 +119,7 @@ public InstanceGoogleSheetsUploader(GoogleAccountCredential credential) {
                 .build();
     }
 
-    protected void uploadInstances(String selection, String[] selectionArgs, String token) {
+    protected void uploadInstances(String selection, String[] selectionArgs, String token, int low, int vauesCount) {
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/tasks/InstanceGoogleSheetsUploader.java]: `vauesCount` should be `instanceCount` I think.

**Review by lognaturel — COMMENTED** [review]
Thanks! One small naming change and I think this is ready to go.

**Comment by grzesiek2010** [comment]
@lognaturel fixed.

**Pull request merged**